src/eventLogger.ts

```diff
@@ -0,0 +1,47 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ramiro González Maciel <ramiro@modusbox.com>
+
+ --------------
+ ******/
+
+'use strict'
+
+import { MessageType } from "./model/MessageType";
+
+
+/**
+ * SDK Client - NOT FINAL
+*/
+class EventLogger {
```

Comment:
A couple of comments here. 

Please define this as an interface or abstract class in addition that is inherited by this class. 

The reason for this is that we want to include crypto-sigs etc going forward. As a result, I think it would be great if we can define an abstract class with place-holder methods that people can extend and then rather use dependency injection to create the EventLogger.

Have a think about how we can do something along these lines that would give us and users of the SDK flexibility going forward.

Reply:
See [feature/#803ClientAndServerImpl 3acddd2] Defines EventLogger as an interface, adds EventPreProcessor and EventPostProcessor interfaces